### PR TITLE
First draft of OAuth documentation

### LINKED ISSUE #942: Document SaaS connector OAuth steps

### PR DESCRIPTION
**Pull request by galvana**
# Purpose
To document the `oauth2` authentication strategy and how OAuth2 is configured at the Fidesops instance level.

# Changes
- Added the **SaaS OAuth2 Configuration** section in the documentation.

# Checklist
- [x] Update [`CHANGELOG.md`](https://github.com/ethyca/fidesops/blob/main/CHANGELOG.md) file
  - [x] Merge in main so the most recent `CHANGELOG.md` file is being appended to
  - [x] Add description within the `Unreleased` section in an appropriate category. Add a new category from the list at the top of the file if the needed one isn't already there.
  - [x] Add a link to this PR at the end of the description with the PR number as the text. example: [#1](https://github.com/ethyca/fidesops/pull/1)
- [x] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md).
- If docs updated (select one):
  - [x] documentation complete, or draft/outline provided (tag docs-team to complete/review on this branch)

# Ticket

Fixes #942

### CODE REVIEW

File: docs/fidesops/mkdocs.yml [hunk 1]
```diff
@@ -3,56 +3,57 @@ site_url: https://ethyca.github.io/fidesops/
 nav:
   - What is Fidesops?: index.md
   - Use Cases: use_cases.md
-  - Installation and Use:
```
Comment (galvana) [inline docs/fidesops/mkdocs.yml:6]: Whitespace changes from auto-formatting.

File: docs/fidesops/docs/saas_connectors/saas_connectors.md [hunk 2]
```diff
@@ -50,7 +50,7 @@ PATCH api/v1/connection/{saas_key}/saas_config
 ```
 3. Configure the secrets. The SaaS config must already defined to provide validation for the secrets.
 ```
-PUT api/v1/connection/{saas_key}/secret?verify=true
```
Comment (galvana) [inline docs/fidesops/docs/saas_connectors/saas_connectors.md:53]: I refer to these instructions from the OAuth documentation and we can't verify right away for OAuth2 connectors so I removed `verify=true` so these instructions apply to both OAuth2 and non-OAuth2 connectors.

File: docs/fidesops/docs/saas_connectors/saas_connectors.md [hunk 3]
```diff
@@ -8,14 +8,14 @@ A SaaS (Software as a Service) connection is a connection type within fidesops t
 
 The current implementation of the SaaS framework can support any SaaS application that uses these features:
 
-- Basic and bearer authentication
-- Data access via HTTP GET requests
-- Erasure via HTTP PUT requests
+- Basic auth, bearer auth, OAuth2 (Authorization Code Flow)
```
Comment (galvana) [inline docs/fidesops/docs/saas_connectors/saas_connectors.md:11]: Updating the features we support.

**Comment by seanpreston** [comment]
Thanks for adding this @galvana — it looks mostly good. We'll probably want to get the design folks to give us a polished version of that diagram.

**Review by conceptualshark — APPROVED** [review]
Thanks for setting this up. 👍 I made a few small edits for clarity/style, but other than needing a final diagram, this looks good to me!

**Pull request merged**